### PR TITLE
report: metric filter refactor to JS and adornments

### PR DESCRIPTION
**Pull request by paulirish**
some (potential) followups from #11732

**JS refactor:** After [connor raised the question](https://github.com/GoogleChrome/lighthouse/pull/11732#discussion_r606377369), I tried out the JS approach to see. I'm fine with either, so if folks prefer the JS style, wfm.

**adornments [idea](https://github.com/GoogleChrome/lighthouse/pull/11732#issuecomment-814359015)**: the data side of it was annoying enough to split into this. The UI bit seems interesting, no strong feelings, tbh.

demo: https://lighthouse-git-adorn-googlechrome.vercel.app/english/

### CODE REVIEW

**Comment by connorjclark** [comment]
In both demo reports, selecting "CLS" removes any relevant audits from the "Passed Audits" section. Ideally that entire section would be hidden while in this state. Same for "Opportunities". Can that be done with just CSS? Probably, because it's not just CSS, you're dynamically writing whatever CSS you want with JS.

For the report, something weird is happening when you toggle to CLS and then back to All–there is then nothing left in "Passed Audits".

I still think that JS is simpler to understand than dynamically generated CSS.

**Comment by paulirish** [comment]
> selecting "CLS" removes any relevant audits from the "Passed Audits" section. Ideally that entire section would be hidden while in this state. 

yeah i was skipping that functionality when this was CSS only. now that it's JS-driven i can add it.


> For this PR's demo, something weird is happening when you toggle to CLS and then back to All–there is then nothing left in "Passed Audits".

good catch. i'll fix that.

File: lighthouse-core/report/html/renderer/performance-category-renderer.js [hunk 1]
```diff
@@ -301,66 +301,63 @@ class PerformanceCategoryRenderer extends CategoryRenderer {
...
-/* …And then display:block the relevant ones */
-${auditSelectors.join(',\n')} {
-  display: block;
-}
-/*# sourceURL=metricfilter.css */
-    `;
-    categoryEl.append(metricFilterEl);
+      // Toggle class/hidden state based on filter choice.
+      radioEl.addEventListener('input', _ => {
+        categoryEl.querySelectorAll('label.lh-metricfilter__label').forEach(elem => {
```
Comment (connorjclark) [inline lighthouse-core/report/html/renderer/performance-category-renderer.js:334]: nit: you do forEach and a for of loop here. pick one? I suggest for of. using forEach is awkward imo and unless you need the extra args

File: lighthouse-core/test/results/sample_v2.json [hunk 2]
```diff
@@ -7072,6 +7072,7 @@
       "runtimeSettingsUANetwork": "User agent (network)",
       "runtimeSettingsUrl": "URL",
       "runtimeUnknown": "Unknown",
+      "showRelevantAudits": "Show audits relevant to: ",
```
Comment (connorjclark) [inline lighthouse-core/test/results/sample_v2.json:7075]:
```suggestion
      "showRelevantAudits": "Show audits relevant to:",
```

I think? we can just do a space in html

**Review by connorjclark — APPROVED** [review]

**Pull request merged**